src/posit/connect/_repository.py

```diff
@@ -0,0 +1,145 @@
+"""Content item repository."""
+
+from __future__ import annotations
+
+from abc import abstractmethod
+
+from typing_extensions import (
+    Any,
+    Hashable,
+    Optional,
+    Protocol,
+    overload,
+)
+
+from .errors import ClientError
+from .resources import Resource, _Resource
+
+
+# TODO-barret: Replace with Resource class from https://github.com/posit-dev/posit-sdk-py/pull/364/files#diff-94b7dc3c7d7d7c7b1a5f25e06c37df5fc53e1921cb10d41d4f04b18a715fae55R72
+class ResourceP(Protocol):
+    def __getitem__(self, key: Hashable, /) -> Any: ...
+
+
+# ContentItem Repository uses a PATCH method, not a PUT for updating.
+class _ContentItemRepository(_Resource):
+    def update(self, **attributes):
+        response = self._ctx.client.patch(self._path, json=attributes)
+        result = response.json()
+        # # Calling this method will call `_Resource.update` which will try to PUT to the path.
+        # super().update(**result)
+        # Instead, update the dict directly.
+        dict.update(self, **result)
```

Comment:
This might be controversial, but I couldn't find a way to skip `_Resource`'s `.update()` method and only perform the `dict`'s update. 

By passing in self to `dict.update(self, **result)`, I was able to have full control over what was being performed. 

Happy to use a different solution, but I couldn't come up with a clean approach.

Reply:
I agree it's unusual, nevertheless it's correct.

Another option is... This may be easier to understand, but it isn't as simple. I don't have strong opinions either way. If you decide to keep the implementation above, then you could add the following snippet to give the reader a quicker understanding of what the method is doing.

```
        result: dict = response.json()
        for key, value in result.items():
            self[key] = value
```